### PR TITLE
Add pyorc 0.4.0

### PR DESCRIPTION
**Pull request by noirello**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by noirello** [comment]
@conda-forge/help-python-c

**Comment by noirello** [comment]
@conda-forge-admin, please ping team

Sorry for bothering, but I'm not sure why I'm unable to install pybind11.

File: recipes/pyorc/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    {{ PYTHON }} setup.py build_orc --build-type=release
+    {{ PYTHON }} -m pip install . -vv
+  skip: True  # [py<36]
+
+requirements:
+  build:
+    - {{ compiler('cxx') }}
+  host:
+    - python
+    - pip
```
Comment (chrisburr) [inline recipes/pyorc/meta.yaml:25]:
```suggestion
    - pip
    - pybind11
```

**Review by isuruf — DISMISSED** [review]
Please package the license/copright notice of `orc` in addition to the license of `pyorc`

File: recipes/pyorc/meta.yaml (hunk 1 above)
Comment (noirello) [reply recipes/pyorc/meta.yaml:24]: Thank you.

File: recipes/pyorc/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+requirements:
+  build:
+    - {{ compiler('cxx') }}
+    - cmake
+    - make  # [not win]
+    - ninja  # [win]
+  host:
+    - python
+    - pip
+    - pybind11
```
Comment (chrisburr) [inline recipes/pyorc/meta.yaml:28]: There might be others but it looks like the orc build needs some dependencies adding:
```suggestion
    - pybind11
    - snappy
    - zstd
    - zlib
```

Comment (noirello) [reply recipes/pyorc/meta.yaml:28]: It's going to be something else. These libraries should be built from source during the ORC C++ package install. I'll try to look into it why the snappy build fails.

Comment (xhochy) [reply recipes/pyorc/meta.yaml:28]: Could it be possible to just use the orc package instead of building it here?

Comment (noirello) [reply recipes/pyorc/meta.yaml:28]: I don't think that the ORC package comes with a binary distribution.

Comment (xhochy) [reply recipes/pyorc/meta.yaml:28]: Every conda package is a binary package. `conda install orc` gives you a compiled Apache ORC C++

Comment (noirello) [reply recipes/pyorc/meta.yaml:28]: Thank you, did not know that. Managed to build the python module.

File: recipes/pyorc/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+build:
+  number: 0
+  # If the installation is complex, or different between Unix and Windows, use separate bld.bat and build.sh files instead of this key.
+  # By default, the package will be built for the Python versions supported by conda-forge and for all major OSs.
+  # Add the line "skip: True  # [py<35]" (for example) to limit to Python 3.5 and newer, or "skip: True  # [not win]" to limit to Windows.
+  script: |
+    {{ PYTHON }} setup.py build_orc --download-only=true
+    {{ PYTHON }} -m pip install . -vv
+  skip: True  # [py<36 or win]
```
Comment (xhochy) [inline recipes/pyorc/meta.yaml:17]: We don't build for `py<36`, thus can be omitted here.

```suggestion
  skip: True  # [win]
```

File: recipes/pyorc/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+source:
+  url: https://pypi.io/packages/source/p/pyorc/pyorc-0.4.0.tar.gz
+  sha256: af56c50dd0545b451a37d38e73e02405f332e91779cf9ab1c8ae3e61b9eb0c90
+
+build:
+  number: 0
+  # If the installation is complex, or different between Unix and Windows, use separate bld.bat and build.sh files instead of this key.
+  # By default, the package will be built for the Python versions supported by conda-forge and for all major OSs.
+  # Add the line "skip: True  # [py<35]" (for example) to limit to Python 3.5 and newer, or "skip: True  # [not win]" to limit to Windows.
```
Comment (xhochy) [inline recipes/pyorc/meta.yaml:13]: No comments at the end, please.
```suggestion
```

**Comment by noirello** [comment]
Thank you for all, made the required modifications.

One thing I'm not sure about: does this make the conda-forge ORC lib a runtime dependency? Should I add the orc package to the `run` requirements as well in the manifest?

**Comment by xhochy** [comment]
> Thank you for all, made the required modifications.
> 
> One thing I'm not sure about: does this make the conda-forge ORC lib a runtime dependency? Should I add the orc package to the `run` requirements as well in the manifest?

No need for that as the `orc` has a `run_exports`: https://github.com/conda-forge/orc-feedstock/blob/2fdf03e86ebc155d805a117bfd5f78f9a886c364/recipe/meta.yaml#L21
This means once `orc` is listed in the host section, it will be added on build with the correct version pin to `run`. When a newer version of `orc` is build that is incompatible with the pin, a bot will issue a PR that does a rebuild. I'm going to merge and you will directly see that as we currently do an active rebuild of all packages depending on `orc`.

**Review by xhochy — APPROVED** [review]

**Pull request merged**